### PR TITLE
simplify - fixes issue #21355 and now some equations give correct result

### LINKED ISSUE #21355: `radsimp` gives wrong result for `1 / (x + sqrt(x**2))`
I got into an accidental discovery of this issue while developing #21343
Generally, even if `1 / (x + sqrt(x**2))` can have a nonzero denominator for `x > 0`, the expression is not rationalizable because multiplying `x - sqrt(x**2)` makes the denominator zero. (`x - sqrt(x**2)` should also not be rationalizable.)

I saw that `radsimp(1 / (x + sqrt(x**2)))` gives `zoo*(x + sqrt(x**2))` which should be a wrong result.
I think that it needs a test for intermediate zero or `NaN, zoo`.

### PR DESCRIPTION
**Pull request by praneethratna**
#### Brief description of what is fixed or changed
Fixes #21355 and now `radsimp(1/(x + sqrt(x**2)))` returns `1/(x + sqrt(x**2))` instead of `zoo*(x + sqrt(x**2))` and also added regression test in test_radsimp.py

#### Release Notes

* simplify
   * fixed bug in radsimp.

### CODE REVIEW

**Comment by praneethratna** [comment]
@sylee957 please review

File: sympy/simplify/radsimp.py [hunk 1]
```diff
@@ -966,6 +966,9 @@ def handle(expr):
             n *= num
             d *= num
             d = powdenest(_mexpand(d), force=symbolic)
+            if d.has(S.Zero, nan, zoo):
+                n, d = fraction(expr)
+                return _unevaluated_Mul(n, 1/d)
```
Comment (sylee957) [inline sympy/simplify/radsimp.py:971]: what about `return expr`

Comment (praneethratna) [reply sympy/simplify/radsimp.py]: @sylee957 I've made the changes , any other changes to make?

File: sympy/simplify/radsimp.py [hunk 2]
```diff
@@ -966,6 +966,8 @@ def handle(expr):
             n *= num
             d *= num
             d = powdenest(_mexpand(d), force=symbolic)
+            if d.has(S.Zero, nan, zoo):
```
Comment (smichr) [inline sympy/simplify/radsimp.py:969]: Does the problem already exist with `num` a few lines earlier?

Comment (praneethratna) [reply sympy/simplify/radsimp.py:969]: @smichr No , the problem exists only after line 968.

**Pull request merged**